src/opendap/auth/UrsIdP.java

```diff
@@ -543,9 +543,52 @@ else if (rejectUnsupportedAuthzSchemes) {
                 log.warn(msg, AuthorizationHeader.getScheme(authz_hdr_value));
             }
         }
+        if(foundValidAuthToken) {
+            // Change session id post authentication
+            String newSessionId = request.changeSessionId();
+            log.debug("New Session Id: {}",newSessionId);
+        }
         return foundValidAuthToken;
     }
 
+    /**
+     * Retrieves an EDL authorization token from the EDL SSO by presenting EDL Client Application information and the
+     * auth code provided the query string of the incoming client request (after the client's authentication excursion
+     * to the EDL SSO)
+     * This method is called when the user/client was redirected by URS back to our application,
+     * and we have a code. We now exchange the code for a token, which is returned by the EDL SSO as a json document.
+     *
+     * @param code The EDL auth code from the request url query string
+     * @param returnToUrl The URL to which the EDL SSO will send the client after their authentication attempt.
+     * @return An EarthDataLoginAccessToken object built from the response from the EDL SSO.
+     * @throws IOException
+     */
+    public EarthDataLoginAccessToken exchangeAuthCodeForEdlToken(String code, String returnToUrl) throws IOException {
```

Comment:
...this refactor (pulling this set of code into a new `exchangeAuthCodeForEdlToken` function) is orthogonal to the task at hand, right? Or am I misunderstanding and there's a specific reason to do it as part of this PR? If possible, I'd prefer to do it as a separate follow-up PR, esp since we already know we have some EDL auth cleanup/refactoring to do in the near future.